frontend/src/employee-frontend/components/vasu/VasuStateTransitionButtons.tsx

```diff
@@ -0,0 +1,131 @@
+  text-align: right;
+  padding-top: ${defaultMargins.s};
+`
+
+export function VasuStateTransitionButtons({
+  childId,
+  documentId,
+  state
+}: {
+  childId: UUID
+  documentId: UUID
+  state: VasuDocumentState
+}) {
+  const { i18n } = useTranslation()
+  const history = useHistory()
+
+  const [updateResult, setUpdateResult] = useState<Result<null>>()
+  const updateState = useRestApi(updateDocumentState, setUpdateResult)
+
+  const update = (eventType: VasuDocumentEventType) =>
+    updateState({ documentId, eventType })
+
+  const [
+    selectedEventType,
+    setSelectedEventType
+  ] = useState<VasuDocumentEventType>()
+
+  const getStateTransitionButton = (
+    eventType: VasuDocumentEventType,
+    primary = true
+  ) => (
+    <Button
+      text={i18n.vasu.transitions[eventType].buttonText}
+      onClick={() => setSelectedEventType(eventType)}
+      primary={primary}
+    />
+  )
+
+  return (
+    <div>
+      {selectedEventType && !updateResult?.isSuccess && (
+        <InfoModal
+          iconColour="blue"
+          title={i18n.vasu.transitions[selectedEventType].confirmTitle}
+          icon={faQuestion}
+          text={
+            ['PUBLISHED', 'MOVED_TO_READY', 'MOVED_TO_REVIEWED'].includes(
+              selectedEventType
+            )
+              ? i18n.vasu.transitions.guardiansWillBeNotified
+              : undefined
+          }
+          resolveDisabled={updateResult?.isLoading}
+          resolve={{
+            action: () => update(selectedEventType),
+            label: i18n.vasu.transitions[selectedEventType].confirmAction
+          }}
+          reject={{
+            action: () => setSelectedEventType(undefined),
+            label: i18n.common.goBack
+          }}
+        >
+          {updateResult?.isLoading && <SpinnerSegment />}
+          {updateResult?.isFailure && <ErrorSegment />}
+        </InfoModal>
+      )}
+      {selectedEventType && updateResult?.isSuccess && (
+        <InfoModal
+          iconColour="green"
+          title={i18n.vasu.transitions[selectedEventType].successTitle}
+          icon={faCheck}
+          text={i18n.vasu.transitions[selectedEventType].successText}
+          resolve={{
+            action: () => history.push(`/child-information/${childId}`),
+            label: i18n.common.ok
+          }}
+        />
+      )}
+      <ButtonContainer>
+        {state === 'DRAFT' && getStateTransitionButton('MOVED_TO_READY')}
+        {state === 'READY' && getStateTransitionButton('MOVED_TO_REVIEWED')}
+        {state === 'REVIEWED' && (
+          <RequireRole oneOf={['ADMIN']}>
+            {getStateTransitionButton('MOVED_TO_CLOSED')}
+          </RequireRole>
+        )}
+        {state === 'CLOSED' ? (
+          <RequireRole oneOf={['ADMIN']}>
+            {getStateTransitionButton('RETURNED_TO_READY', false)}
+          </RequireRole>
+        ) : (
+          getStateTransitionButton('PUBLISHED', false)
+        )}
+      </ButtonContainer>
```

Comment:
wondering about the order of buttons. the primary action is usually the last, right? also button for RETURNED_TO_REVIEWED is missing

Reply:
ButtonContainer has `flex-direction: row-reverse;`.

Will add `RETURNED_TO_REVIEWED`.